File: php/features.php
<?php
declare(strict_types=1);

// EarthquakeCatalog SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EarthquakeCatalogFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EarthquakeCatalogBaseFeature();
            case "test":
                return new EarthquakeCatalogTestFeature();
            default:
                return new EarthquakeCatalogBaseFeature();
        }
    }
}

File: py/features.py
# EarthquakeCatalog SDK feature factory

from feature.base_feature import EarthquakeCatalogBaseFeature
from feature.test_feature import EarthquakeCatalogTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EarthquakeCatalogBaseFeature(),
        "test": lambda: EarthquakeCatalogTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
